pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -213,24 +213,28 @@ func garbageCollectChunks(store brtypes.SnapStore, snapList brtypes.SnapList, lo
 	}
 }
 
-// garbageCollectDeltaSnapshots deletes only the delta snapshots from revision sorted <snapStream>. It won't delete the full snapshot
+// GarbageCollectDeltaSnapshots deletes only the delta snapshots from revision sorted <snapStream>. It won't delete the full snapshot
```

Comment:
This function's docstring says the following
```
It won't delete the full snapshot
// in snapstream which supposed to be at index 0 in <snapStream>.
```
However, since the for loop considers the case of `i=0`, this snapshot is in fact considered for deletion.
Please change the loop's condition to `i>0` like it was earlier to ensure this

Alternatively you can also change the docstring to be more accurate

Reply:
Thanks @aaronfern, Improved doc string as part of https://github.com/gardener/etcd-backup-restore/pull/640/commits/736a67bd6ff6eac7739a46ab8169651da9f2be54